### PR TITLE
test: enable e2e documentation test in CI

### PR DESCRIPTION
**Pull request by VoVAllen**
Signed-off-by: Jinjing.Zhou <allenzhou@tensorchord.ai>

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -117,6 +117,10 @@ jobs:
         id: get-latest-tag
       - name: e2e test
         run: make e2e-test
+        env:
+          GIT_LATEST_TAG: ${{ steps.get-latest-tag.outputs.tag }}          
+      - name: e2e doc test
+        run: make doc-test
```
Comment (gaocegege) [inline .github/workflows/CI.yml:123]: Could you please make it parallel?

Comment (gaocegege) [reply .github/workflows/CI.yml:123]: Putting the doc test in another job, instead of an action after e2e.

Comment (VoVAllen) [reply .github/workflows/CI.yml:123]: Sure

File: Makefile [hunk 2]
```diff
@@ -188,7 +188,7 @@ doc-test:
 		-X $(ROOT)/pkg/version.gitTreeState=$(GIT_TREE_STATE)                     \
 		-X $(ROOT)/pkg/version.gitTag="$(shell git describe --tags --abbrev=0)" \
 		-X $(ROOT)/pkg/version.developmentFlag=true" \
-		-race -v -timeout 20m -coverpkg=./pkg/app -coverprofile=e2e-coverage.out ./e2e/docs
```
Comment (gaocegege) [inline Makefile:191]: Could we rename the target `e2e-doc-test`? Since it is part of the e2e test cases.

**Comment by gaocegege** [comment]
Please ping me if it is ready to review.

Thanks.

**Comment by VoVAllen** [comment]
@gaocegege Ready to review

**Review by gaocegege — APPROVED** [review]

**Review by terrytangyuan — APPROVED** [review]

**Comment by gaocegege** [comment]
Let's merge it manually.

**Pull request merged**